### PR TITLE
lib: modem_jwt: update to use new AT APIs

### PR DESCRIPTION
**Pull request by lemrey**
Update library to use new APIs from libmodem.

### CODE REVIEW

**Comment by SeppoTakalo** [comment]
I feel like this really needs some unittests.
Code itself looks OK.

Can we easily stub the AT layer? I can generate some static JWT responses from modem.

**Review by SeppoTakalo — APPROVED** [review]

**Comment by lemrey** [comment]
Yes, the AT APIs can be mocked and return a static response.

**Comment by SeppoTakalo** [comment]
Actually, I just realized that at least this API has a unittest.

```
twister -T ./tests/lib/modem_jwt -p native_posix
```

Fails to build.
At least
```
#include <nrf_modem_at.h>
```
is missing and then in the test case, stubs are wrong, because they stub at_cmd.
Can you fix?

File: lib/modem_jwt/modem_jwt.c [hunk 1]
```diff
@@ -93,71 +86,41 @@ int modem_jwt_generate(struct jwt_data *const jwt)
...
+			JWT_CMD_TEMPLATE_SEC_TAG,
 			jwt->alg,
 			jwt->exp_delta_s,
 			jwt->subject ? jwt->subject : "",
 			jwt->audience ? jwt->audience : "",
 			jwt->sec_tag,
 			jwt->key);
 	} else {
-		ret = snprintf(cmd, cmd_sz, JWT_CMD_TEMPLATE,
+		ret = nrf_modem_at_cmd(cmd_resp, CONFIG_MODEM_JWT_MAX_LEN,
```
Comment (jayteemo) [inline lib/modem_jwt/modem_jwt.c:109]: `nrf_modem_at_cmd` can return a positive value, but `modem_jwt_generate` should not

File: lib/modem_jwt/modem_jwt.c [hunk 2]
```diff
@@ -18,13 +18,6 @@
 #define JWT_CMD_PRINT_INT_SZ (MAX_INT_PRINT_SZ * 4)
 #define GET_BASE64_DECODED_LEN(n) (3 * n / 4)
 
-/* Minimum JWT response is ~420 bytes.
```
Comment (jayteemo) [inline lib/modem_jwt/modem_jwt.c:21]: the information in this comment might be useful in the help text for `MODEM_JWT_MAX_LEN`

**Review by jayteemo — CHANGES_REQUESTED** [review]

**Review by SeppoTakalo — CHANGES_REQUESTED** [review]
This seem to require few changes to properly build the test and modem_jwt.c itself.

```
❯ git diff
  diff --git a/lib/modem_jwt/modem_jwt.c b/lib/modem_jwt/modem_jwt.c
  index c18dbaed2..d02b12035 100644
  --- a/lib/modem_jwt/modem_jwt.c
  +++ b/lib/modem_jwt/modem_jwt.c
  @@ -8,7 +8,7 @@
   #include <zephyr.h>
   #include <stdlib.h>
   #include <stdio.h>
  -#include <modem/at_cmd.h>
  +#include <nrf_modem_at.h>
   #include <modem/modem_jwt.h>
   #include <sys/base64.h>
   
  diff --git a/tests/lib/modem_jwt/CMakeLists.txt b/tests/lib/modem_jwt/CMakeLists.txt
  index ce4c2ceff..0eef59acc 100644
  --- a/tests/lib/modem_jwt/CMakeLists.txt
  +++ b/tests/lib/modem_jwt/CMakeLists.txt
  @@ -14,7 +14,7 @@ test_runner_generate(src/jwt_test.c)
   
   target_include_directories(app PRIVATE src)
   
  -cmock_handle(../../../include/modem/at_cmd.h)
  +cmock_handle(../../../../nrfxlib/nrf_modem/include/nrf_modem_at.h)
   
   # add test file
   target_sources(app PRIVATE src/jwt_test.c)
  diff --git a/tests/lib/modem_jwt/src/jwt_test.c b/tests/lib/modem_jwt/src/jwt_test.c
  index 100a281c5..1d15727ef 100644
  --- a/tests/lib/modem_jwt/src/jwt_test.c
  +++ b/tests/lib/modem_jwt/src/jwt_test.c
  @@ -9,7 +9,7 @@
   #include <string.h>
   #include <kernel.h>
   #include <modem/modem_jwt.h>
  -#include <mock_at_cmd.h>
  +#include <mock_nrf_modem_at.h>
   
   
   void setUp(void)
  @@ -37,19 +37,15 @@ static const char jwt_token[] =
          "eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NiIsImtpZCI6ImU2YzM1Y2VhMWY1NDFhNGJlNDg1ZmY4ZGIyMWY1MWRmZTcyNDJlN2NhN2I5NDU2YzkwZjY5ODE1ZTc3ZWE2MWYifSAg.eyJpc3MiOiJuUkY5MTYwLjUwMzYzMTU0LTM5MzAtNGQ5Ni04MGY5LTEzMTIxY2YzNWFkYyIsImp0aSI6Im5SRjkxNjAuZjcyZDk3ODktMm  U3My00NzgwLTkyNDgtYjM5NTJmODliODcyLjk0MmQ5OWI5MDNhY2EyYzNkZDY0OWI1ZDYxMjMyMmRjIiwic3ViIjoiNTAzNjMxNTQtMzkzMC00ZDk2LTgwZjktMTMxMjFjZjM1YWRjIn0g.tW2wG-sSE01jtrcT9JmFvPVIvEB99jUGGHIUCCH65fG5HtBrisOD1cl0L49cuyo2OE6XXvI9wdKSASxDQPJ8lw"
          "\"";
   
  -int cmd_cb(const char * const cmd, char *buf, size_t buf_len, enum at_cmd_state *state,
  -          int cmock_num_calls)
  +int cmd_cb(void* buf, size_t len, const char* fmt, int cmock_num_calls)
   {
          if (cmock_num_calls == 0) {
                  /* Fail first call */
  -               return AT_CMD_ERROR;
  -       } else if (cmock_num_calls == 1) {
  -               TEST_ASSERT_EQUAL_STRING("AT%JWT=0,0,\"\",\"\"", cmd);
  -       } else if (cmock_num_calls == 2) {
  -               TEST_ASSERT_EQUAL_STRING("AT%JWT=0,100,\"sub\",\"aud\",10,2", cmd);
  +               return -1;
          }
  -       strncpy(buf, jwt_token, buf_len - 1);
  -       buf[buf_len-1] = 0;
  +
  +       strncpy(buf, jwt_token, len - 1);
  +       ((char *)buf)[len-1] = 0;
          return 0;
   }
   
  @@ -70,17 +66,14 @@ void test_modem_jwt_generate(void)
   
          jwt.jwt_buf = NULL;
   
  -       __wrap_at_cmd_init_ExpectAndReturn(-1);
  -       rc = modem_jwt_generate(&jwt);
  -       TEST_ASSERT_EQUAL_INT(-EPIPE, rc);
  +       __wrap_nrf_modem_at_cmd_Stub(cmd_cb);
   
  -       __wrap_at_cmd_init_ExpectAndReturn(0);
  -       __wrap_at_cmd_write_Stub(cmd_cb);
  +       /* First call should fail, pass through the error code */
          rc = modem_jwt_generate(&jwt);
  -       TEST_ASSERT_EQUAL_INT(-EBADMSG, rc);
  +       TEST_ASSERT_EQUAL_INT(-1, rc);
  +
   
          /* Success case, default parameters */
  -       __wrap_at_cmd_init_ExpectAndReturn(0);
          rc = modem_jwt_generate(&jwt);
          TEST_ASSERT_EQUAL_INT(0, rc);
   
  @@ -93,7 +86,6 @@ void test_modem_jwt_generate(void)
          jwt.exp_delta_s = 100;
          jwt.sec_tag = 10;
          jwt.key = 2;
  -       __wrap_at_cmd_init_ExpectAndReturn(0);
          rc = modem_jwt_generate(&jwt);
          TEST_ASSERT_EQUAL_INT(0, rc);
   }
  @@ -104,7 +96,6 @@ void test_modem_jwt_get_uuids(void)
          struct nrf_modem_fw_uuid mfw = {0};
          int rc;
   
  -       __wrap_at_cmd_init_IgnoreAndReturn(0);
          rc = modem_jwt_get_uuids(&dev, &mfw);
          TEST_ASSERT_EQUAL_INT(0, rc);
          TEST_ASSERT_EQUAL_STRING("50363154-3930-4d96-80f9-13121cf35adc", dev.str);

```

**Comment by lemrey** [comment]
Hey @SeppoTakalo if you have fixed the unit tests feel free to push a commit to my branch :)

File: lib/modem_jwt/modem_jwt.c (hunk 1 above)
Comment (lemrey) [reply lib/modem_jwt/modem_jwt.c:109]: Fixed

**Comment by lemrey** [comment]
Thanks a lot @SeppoTakalo , I have squashed your commit.

File: lib/modem_jwt/modem_jwt.c (hunk 2 above)
Comment (lemrey) [reply lib/modem_jwt/modem_jwt.c:21]: Added to help text

**Review by SeppoTakalo — APPROVED** [review]

**Review by jayteemo — APPROVED** [review]

**Pull request merged**